pyartcd/pyartcd/pipelines/olm_bundle.py

```diff
@@ -0,0 +1,87 @@
+from pyartcd.runtime import Runtime
+
+
+@cli.command('olm-bundle')
+@click.option('--version', required=True, help='OCP version')
+@click.option('--assembly', required=True, help='Assembly name')
+@click.option('--data-path', required=False, default=constants.OCP_BUILD_DATA_URL,
+              help='ocp-build-data fork to use (e.g. assembly definition in your own fork)')
+@click.option('--data-gitref', required=False,
+              help='(Optional) Doozer data path git [branch / tag / sha] to use')
+@click.option('--nvrs', required=False,
+              help='(Optional) List **only** the operator NVRs you want to build bundles for, everything else gets ignored. The operators should not be mode:disabled/wip in ocp-build-data')
+@click.option('--only', required=False,
+              help='(Optional) List **only** the operators you want to build, everything else gets ignored.\n' 
+                   'Format: Comma and/or space separated list of brew packages (e.g.: cluster-nfd-operator-container)\n'
+                   'Leave empty to build all (except EXCLUDE, if defined)')
+@click.option('--exclude', required=False,
+              help='(Optional) List the operators you **don\'t** want to build, everything else gets built.\n'
+                   'Format: Comma and/or space separated list of brew packages (e.g.: cluster-nfd-operator-container)\n'
+                   'Leave empty to build all (or ONLY, if defined)')
+@click.option('--force', is_flag=True,
+              help='Rebuild bundle containers, even if they already exist for given operator NVRs')
+@pass_runtime
+@click_coroutine
+async def olm_bundle(runtime: Runtime, version: str, assembly: str, data_path: str, data_gitref: str,
+                     nvrs: str, only: bool, exclude: str, force: bool):
+    # Create Doozer invocation
+    cmd = [
+        'doozer',
+        f'--assembly={assembly}',
+        '--working-dir=doozer_working',
+        f'--group=openshift-{version}@${data_gitref}' if data_gitref else f'--group=openshift-{version}',
+        f'--data-path={data_path}'
+    ]
+    if only:
+        cmd.append(f'--images={only}')
+    if exclude:
+        cmd.append(f'--exclude={exclude}')
+    cmd.append('olm-bundle:rebase-and-build')
+    if force:
+        cmd.append('--force')
+    if runtime.dry_run:
+        cmd.append('--dry-run')
+    cmd.extend(
+        [
+            '--',
+            ' '.join(nvrs.split(','))
+        ]
+    )
+
+    # Create a Lock manager instance
+    redis_url = locks.redis.substitute(
+        redis_password=os.environ['REDIS_SERVER_PASSWORD'],
+        redis_host=os.environ['REDIS_HOST'],
+        redis_port=os.environ['REDIS_PORT']
+    )
+    retry_policy = locks.RETRY_POLICY['olm_bundle']
+    lock_manager = Aioredlock(
+        [redis_url],
+        internal_lock_timeout=locks.LOCK_TIMEOUTS['olm-bundle'],
+        retry_count=retry_policy['retry_count'],
+        retry_delay_min=retry_policy['retry_delay_min']
+    )
```

Comment:
It would nice if these lines are moved to a separate function (e.g. `new_lock_manager` in `locks.py`) so that we can reuse it in the future.

Reply:
Very good idea, implemented in 96ea8bc88de32d9438e862561ad40619005862fe